### DISCUSSION TITLE
Error: Add to Calendar Button generation (atcb-btn-1) failed: invalid option wordpress

### DISCUSSION THREAD

**Discussion by jaycova** [add2cal/add-to-calendar-button#497] [Q&A] [answered]
I am having an issue when I install Add to Calendar Button. I am getting the following error:

Error: Add to Calendar Button generation (atcb-btn-1) failed: invalid option WordPress

I am using WordPress version 6.3.2
PHP version 8.1.10

I get the error as soon as I start changing the start date, and even if I don't change anything, it gets this error: 

Would anyone happen to have any ideas? It is a great little plugin once I get it working.

**Comment by jekuer** [comment]
Are you using the shortcode or Gutenberg block?

**Reply by jaycova** [reply]
Gutenberg Block

**Comment by jaycova** [comment]
So I just tried it with shortcode, and it seems to work.

**Comment by jekuer** [comment]
Would you mind to share a screenshot of the Block settings?
This error message has to do with the calendar type options. However, seeing "WordPress" as a string, should only be possible if you try to specify this as an option and only if done via the "other attributes" block.

**Reply by jaycova** [reply]
![Screenshot 2023-11-06 at 1 51 08 PM](https://github.com/add2cal/add-to-calendar-button/assets/8138115/6dcd352f-aff2-4b46-aeec-69108cfc5236)
is this what you need?

**Reply by jekuer** [reply]
Yes. You need to select an option at "calendar options".
Unfortunately, this selection clears each time you change "other attributes" - we are aware of this and will optimize soon. Sorry for this inconvenience.
You can select multiple options holding Ctrl.

**Comment by jaycova** [comment]
![Screenshot 2023-11-06 at 1 51 08 PM](https://github.com/add2cal/add-to-calendar-button/assets/8138115/26ba3536-4509-4586-ad6b-c688aa2a2f91)

So when I choose Apple, I get the error attached as the date is misspelled.

**Reply by jaycova** [reply]
sorry wrong image 
![this one](https://github.com/add2cal/add-to-calendar-button/assets/8138115/65782e3e-de19-4576-bb8a-73927acd87bf)

**Reply by jekuer** [reply]
It is hard to tell by the image, but it seems to look like you copy+paste content and accidentally copied a special character instead if the "real" one.
Try deleting the start date and rewrite it by hand again.

**Reply by jaycova** [reply]
No, the highlighted part was because I was trying to cut and paste the error for you. However, I did as you suggested, and it still has an issue. 

It does seem to be an issue between the plugin and Gutenberg blocks, as I got the shortcode working without issues.  you can view the working one at https://www.jaycocioservices.com/webinar_events/ .

**Reply by jekuer** [reply]
Super strange.
Which version of the plugin are you using and which plugin are you using to build the page (I was wondering what the wrapping block type is)?

**Reply by jaycova** [reply]
I am using version 1.5.1. Also, I noticed the add to calendar button does not show on my iPhone browser.

**Reply by jekuer** [reply]
Thanks a lot.
To dig a little deeper: Would it be possible for you to share a screenshot of how the button code looks in the browser dev tools?
Could be that another plugin is sneaking in some code.

**Reply by jaycova** [reply]
This is what you wanted. I am more than happy to give you what you need. 
![source code](https://github.com/add2cal/add-to-calendar-button/assets/8138115/7f6be5c3-ff85-423d-acc7-663700115a7a)

**Reply by jekuer** [reply]
Almost. Go to the "Elements" tab and try to find the button.
Or temporarily add it to the page you linked (or another more hidden one) - then I can check it myself.

**Comment by jaycova** [comment]
Ok. I updated the plugin and the WordPress to 6.4. Here is the screenshot. 
![screenshot1](https://github.com/add2cal/add-to-calendar-button/assets/8138115/ad3720f2-ea31-4752-bd58-9a5afff81680)

**Reply by jekuer** [reply]
Ok, this pretty much supports my first assumption. At the `startDate`, you are not using regular quotes! Those are somehow special characters, which usually get in, if you copy+paste from somewhere else, where something rewrote the regular quotes (e.g. MS Word does so). If it is not the quotes, then maybe the minus in the date.

If you copy the text from the shortcode into the block, it should work. Or alternatively the following one:
```
startDate="2024-11-09"
location="World Wide Web"
```

**Reply by jaycova** [reply]
Ok. This is crazy as I am just using the " from my Mac. Not sure why but I cut and pasted what you typed, and it works now. But both the shortcode and the Gutenberg block do not show on my phone.

**Reply by jekuer** [reply]
Regarding phone - see comment below.

**Comment by jaycova** [comment]
BTW while the Add To Calendar button on the shortcode does work on my MAC Safari Browse. It does not show on my iPhone Safari browser but, 1 issue at a time :)

**Reply by jekuer** [reply, marked as answer]
This is most probably, because of another plugin. When checking your website on iOS, I saw that it got transformed into an AMP-page. AMP does not work nicely together with the button or any similar javascript based tool - at least when using a plugin, which does some random rewriting of the source code. 

Maybe your AMP plugin offers some way to include things like the button, but not sure and doubt it.

**Comment by jaycova** [comment]
Once again you are correct. I turned off amp and there it is.  Amp does not seem to be worth it. Sent from my iPhoneOn Nov 8, 2023, at 12:19 PM, Jens Kuerschner ***@***.***> wrote:﻿
This is most probably, because of another plugin. When checking your website on iOS, I saw that it got transformed into an AMP-page. AMP does not work nicely together with the button or any similar javascript based tool - at least when using a plugin, which does some random rewriting of the source code.
